### PR TITLE
String searching with wildcard

### PR DESCRIPTION
**Pull request by angy113**
@carledriss ready to code review

### CODE REVIEW

File: String searching with wildcard/main/java/org/fundacionjala/StringSearching/SearchEngine.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     * The searching is realized following a pattern that replaces all
+     * the "_" (wildcard) character with any letter.
+     *
+     * @param needle   Is the word with wildcards ("_") to search.
+     * @param haystack Is the string where to find coincidences.
+     * @return The position where the coincidence was found, otherwise
+     * returns -1.
+     */
+    public static int find(String needle, String haystack) {
+        String regNeedle = "\\Q" + needle.replaceAll("_", "\\\\E.\\\\Q") + "\\E";
```
Comment (carledriss) [inline String searching with wildcard/main/java/org/fundacionjala/StringSearching/SearchEngine.java]: this should be final

File: String searching with wildcard/test/java/org/fundacionjala/StringSearching/SearchEngineTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for {@link SearchEngine}
+ */
+public class SearchEngineTest {
+    private final String haystack = "Once upon a midnight dreary, while I pondered, weak and weary";
```
Comment (carledriss) [inline String searching with wildcard/test/java/org/fundacionjala/StringSearching/SearchEngineTest.java]: this should be a constant
private static final

File: String searching with wildcard/main/java/org/fundacionjala/StringSearching/SearchEngine.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package org.fundacionjala.StringSearching;
```
Comment (carledriss) [inline String searching with wildcard/main/java/org/fundacionjala/StringSearching/SearchEngine.java]: Remember
the package name should be lowercase.

**Comment by angy113** [comment]
@carledriss ready to code review

**Comment by carledriss** [comment]
Looks good to me!

**Pull request merged**